src/launchpad/components/Result.js

```diff
@@ -30,9 +26,12 @@ const Result = React.createClass({
   },
 
   onHeaderClick: function() {
-    this.setState(prevState => ({
-      showPacket: !prevState || !prevState.showPacket
-    }));
+    const {expression} = this.props;
+    if (expression.showPacket === true) {
```

Comment:
hehe - in cases like this, I think it's okay to do `if (expression.showPacket) {}` I'm not as paranoid as others. Up to you though